tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-with-query-construct-type.bal

```diff
@@ -1827,6 +1869,40 @@ function testInnerQueryConstructedWithCEP() {
     assertEqual([["01",["func1",["foo",["int","string"],"boolean"]]],["02"]], decl);
 }
 
+error onConflictError = error("Key Conflict", message = "cannot insert.");
+
+function testTableConstructQueryWithNonConflictingKeys() {
+    Customer c1 = {id: 1, name: "Melina", noOfItems: 12};
+    Customer c2 = {id: 2, name: "James", noOfItems: 5};
+    Customer c3 = {id: 3, name: "Anne", noOfItems: 20};
+    Customer[] customerList = [c1, c2, c3];
+    CustomerTable|error customerTable = getQueryResult(onConflictError, customerList);
+    assertEqual(true, customerTable is CustomerTable);
+    CustomerTable expectedTableValue = table [{id: 1, name: "Melina", noOfItems: 12},
+                             {id: 2, name: "James", noOfItems: 5},
+                             {id: 3, name: "Anne", noOfItems: 20}];
+    assertEqual(customerTable, expectedTableValue);
+}
+
+function testTableConstructQueryWithConflictingKeys() {
+    Customer c1 = {id: 1, name: "Melina", noOfItems: 12};
+    Customer c2 = {id: 2, name: "James", noOfItems: 5};
+    Customer c3 = {id: 3, name: "Anne", noOfItems: 20};
+    Customer[] customerList = [c1, c2, c3, c1];
+    CustomerTable|error customerTable = getQueryResult(onConflictError, customerList);
+    assertEqual(customerTable is error, true);
+    assertEqual((<error>customerTable).message(), "Key Conflict");
+}
+
+function getQueryResult(error onConflictError, Customer[] customerList) returns CustomerTable|error {
+    return <CustomerTable|error> table key(id, name) from var customer in customerList
+    select {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+    }
+    on conflict onConflictError;
+}
```

Comment:
Shall we add few tests with `Query actions` as well.

Reply:
No need to add tests for query actions. OnConflict clause applies only when creating tables and maps.